sdk/nodejs/src/client.ts

```diff
@@ -98,6 +105,8 @@ export class DVCClient {
     }
 
     variable<T extends DVCVariableValue>(user: DVCUser, key: string, defaultValue: T): DVCVariable<T> {
+        const incomingUser = castIncomingUser(user)
```

Comment:
what does separating out the user into another class do for us if it does the same thing?

Reply:
it aligns it with the other SDKs so that you can call the sdk like:

```
const user = new DVCUser({user_id: 'user'})
const variable = dvcClient.variable(user, 'key', false)
```

while maintaining backwards compatibility. Basically the other SDKs all do their user data validation inside the DVCUser class rather than within each client call, so the test harness tests can't be written the same way